include/util/json_renderer.hpp

```diff
@@ -4,191 +4,165 @@
 #ifndef JSON_RENDERER_HPP
 #define JSON_RENDERER_HPP
 
-#include "util/cast.hpp"
-#include "util/ieee754.hpp"
 #include "util/string_util.hpp"
 
 #include "osrm/json_container.hpp"
 
+#include <algorithm>
 #include <iterator>
 #include <ostream>
 #include <string>
 #include <vector>
 
+#include <fmt/compile.h>
+
 namespace osrm
 {
 namespace util
 {
 namespace json
 {
 
-namespace
-{
-constexpr int MAX_FLOAT_STRING_LENGTH = 256;
-}
-
-struct Renderer
+template <typename Out> struct Renderer
 {
-    explicit Renderer(std::ostream &_out) : out(_out) {}
+    explicit Renderer(Out &_out) : out(_out) {}
 
-    void operator()(const String &string) const
+    void operator()(const String &string)
     {
-        out << "\"";
-        out << escape_JSON(string.value);
-        out << "\"";
-    }
-
-    void operator()(const Number &number) const
-    {
-        char buffer[MAX_FLOAT_STRING_LENGTH] = {'\0'};
-        ieee754::dtoa_milo(number.value, buffer);
+        write('"');
+        // here we assume that vast majority of strings don't need to be escaped,
+        // so we check it first and escape only if needed
+        if (RequiresJSONStringEscaping(string.value))
```

Comment:
We probably could return index from this function and start `EscapeJSONString` from it, but IMO it is overkill.